src/ui/components/results/resultsheadercomponent.js

```diff
@@ -34,40 +38,91 @@ export default class ResultsHeaderComponent extends Component {
     this.resultsLength = data.resultsLength || 0;
 
     /**
-     * Array of filterNodes to display in the applied filters bar.
+     * Array of potentially removable filterNodes to display.
+     * If config.removable is set as false these will not render as removable.
      * @type {Array<FilterNode>}
      */
-    this.appliedFilterNodes = data.appliedFilterNodes || [];
+    this.removableFilterNodes = data.removableFilterNodes || [];
+
+    /**
+     * Array of irremovable filterNodes to display.
+     * These will not render as removable even if config.removable is true.
+     * @type {Array<FilterNode>}
+     */
+    this.irremovableFilterNodes = data.irremovableFilterNodes || [];
   }
 
   static areDuplicateNamesAllowed () {
     return true;
   }
 
+  onMount () {
+    const removableFilterTags = DOM.queryAll(this._container, '.js-yxt-ResultsHeader-removableFilterTag');
+    removableFilterTags.forEach(tag => {
+      DOM.on(tag, 'click', () => this._removeFilterTag(tag));
+    });
+  }
+
+  _removeFilterTag (tag) {
+    const { filterId } = tag.dataset;
+    const simpleFilterNode = this.removableFilterNodes[filterId];
+    simpleFilterNode.remove();
```

Comment:
This feels kinda weird to me. I think the FilterNodeRegistry should have some kind of remove method that we can call. A registry suggests you an add or delete FilterNodes.

Reply:
maybe a better flow might be to give every filterNode (not just the top level ones) a filterId like we talked about, and call a this.core.removeFilterNode(filterId)? I didn't really want to do this right now but if you think it's necessary I can

Reply:
will resolve in next pr